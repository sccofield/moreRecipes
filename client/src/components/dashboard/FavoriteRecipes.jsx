import React from 'react';
import ReactPaginate from 'react-paginate';
import moment from 'moment';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';

const FavoriteRecipes = props => (
  <div>

    {props.favouriteRecipes.length === 0 ? (
      <p>You have not added any recipe to your favourites yet.</p>
    ) : (


      <section className="section text-center text-lg-left container">
        <nav className="row justify-content-center">
          <ReactPaginate
            previousLabel="previous"
            nextLabel="next"
            breakLabel={<a>...</a>}
            breakClassName="break-me"
            pageCount={props.pageCount}
            marginPagesDisplayed={2}
            pageRangeDisplayed={5}
            onPageChange={props.handlePageClick}
            containerClassName="pagination"
            subContainerClassName="pages pagination"
            activeClassName="active"
            pageClassName="page-item"
            pageLinkClassName="page-link"
            nextLinkClassName="page-link"
            previousLinkClassName="page-link"
          />
        </nav>

        {props.favouriteRecipes.map(favouriteRecipe => (

          <div key={favouriteRecipe.id}>

            <div className="row">


              <div className="col-lg-4 mb-4">

                <div className="view overlay hm-white-slight z-depth-1">
                  <img
                    alt=""
                    src={favouriteRecipe.Recipe.imageUrl}
                    className="profileImage"
                  />
                  <a>
                    <div className="mask" />
                  </a>
                </div>
              </div>

              <div className="col-lg-7">

                <h4 className="">
                  <Link to={`recipes/${favouriteRecipe.Recipe.id}`}>
                    <strong>{favouriteRecipe.Recipe.title}</strong>
                  </Link>
                </h4>
                <p>{
                  favouriteRecipe.Recipe.description.slice(0, 80).concat(' ...')
                }
                </p>
                <p>by
                  <a>
                    <strong> {favouriteRecipe.Recipe.userName}</strong>
                  </a>, {moment(favouriteRecipe.Recipe.createdAt).format('ll')}

                </p>


              </div>


            </div>

            <hr />
          </div>
        ))}
      </section>
    )}
  </div>
);

FavoriteRecipes.propTypes = {
  favouriteRecipes: PropTypes.arrayOf(PropTypes.object).isRequired,
  pageCount: PropTypes.number,
  handlePageClick: PropTypes.func.isRequired
};
FavoriteRecipes.defaultProps = {
  pageCount: null
};

export default FavoriteRecipes;
