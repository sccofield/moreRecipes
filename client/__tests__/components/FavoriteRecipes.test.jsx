import React from 'react';
import toJSON from 'enzyme-to-json';
import FavoriteRecipes from '../../src/components/dashboard/FavoriteRecipes';

describe('Recipe component', () => {
  const props = {
    handlePageClick: jest.fn(() => Promise.resolve()),
    pageCount: 2,
    favouriteRecipes: [{
      id: 3,
      Recipe: {
        title: 'title',
        imageUrl: 'image',
        description: 'description',
        userName: 'username',
        createdAt: 'createdat'

      }

    }]
  };

  const noFavouriteProps = {
    handlePageClick: jest.fn(() => Promise.resolve()),
    pageCount: 2,
    favouriteRecipes: []
  };


  it('should render FavoriteRecipes form when props is given', () => {
    const wrapper = shallow(<FavoriteRecipes {...props} />);
    expect(wrapper.find('div').length).toEqual(7);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render text when props is not given', () => {
    const wrapper = shallow(<FavoriteRecipes {...noFavouriteProps} />);
    expect(wrapper.find('p').length).toEqual(1);
    expect(wrapper.find('p')
      .text()).toBe('You have not added any recipe to your favourites yet.');
  });
});

