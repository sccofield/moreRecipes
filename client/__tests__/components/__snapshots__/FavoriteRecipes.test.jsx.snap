// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component should render FavoriteRecipes form when props is given 1`] = `
<div>
  <section
    className="section text-center text-lg-left container"
  >
    <nav
      className="row justify-content-center"
    >
      <PaginationBoxView
        activeClassName="active"
        breakClassName="break-me"
        breakLabel={
          <a>
            ...
          </a>
        }
        containerClassName="pagination"
        disableInitialCallback={false}
        disabledClassName="disabled"
        marginPagesDisplayed={2}
        nextClassName="next"
        nextLabel="next"
        nextLinkClassName="page-link"
        onPageChange={[MockFunction]}
        pageClassName="page-item"
        pageCount={2}
        pageLinkClassName="page-link"
        pageRangeDisplayed={5}
        previousClassName="previous"
        previousLabel="previous"
        previousLinkClassName="page-link"
        subContainerClassName="pages pagination"
      />
    </nav>
    <div
      key="3"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-4 mb-4"
        >
          <div
            className="view overlay hm-white-slight z-depth-1"
          >
            <img
              alt=""
              className="profileImage"
              src="image"
            />
            <a>
              <div
                className="mask"
              />
            </a>
          </div>
        </div>
        <div
          className="col-lg-7 tab-recipe"
        >
          <h4
            className=""
          >
            <Link
              replace={false}
              to="recipes/undefined"
            >
              <strong>
                title
              </strong>
            </Link>
          </h4>
          <p>
            description ...
          </p>
          <p>
            by
            <a>
              <strong>
                 
                username
              </strong>
            </a>
            , 
            Invalid date
          </p>
        </div>
      </div>
      <hr />
    </div>
  </section>
</div>
`;
